pkg/oneinch/fusionorder/auction_details.go

```diff
@@ -72,6 +66,74 @@ type AuctionPoint struct {
 }
 
 type AuctionGasCostInfo struct {
-	GasBumpEstimate  *big.Int
-	GasPriceEstimate *big.Int
+	GasBumpEstimate  int64
+	GasPriceEstimate int64
+}
+
+// DecodeAuctionDetails decodes auction details from hex string
+// ```
+//
+//	struct AuctionDetails {
+//	     bytes3 gasBumpEstimate;
+//	     bytes4 gasPriceEstimate;
+//	     bytes4 auctionStartTime;
+//	     bytes3 auctionDuration;
+//	     bytes3 initialRateBump;
+//	     (bytes3,bytes2)[N] pointsAndTimeDeltas;
+//	 }
+//
+// ```
+// Logic is copied from
+// https://etherscan.io/address/0xfb2809a5314473e1165f6b58018e20ed8f07b840#code#F23#L140
+// nolint: gomnd
+func DecodeAuctionDetails(hexData []byte) (AuctionDetails, error) {
```

Comment:
should check input has enough length for decode operation.

Reply:
I have updated